### DISCUSSION TITLE
What is the preferred development setup for solving issues?

### DISCUSSION THREAD

**Discussion by abhijelly** [moveit/moveit2#1629] [Q&A] [answered]
Hi there,

I've started picking up good first issues and have relied mainly on automated checks on github for testing after pushing my changes. This approach has been working so far because the issues were relatively easy. As I want to start picking up slightly complex issues, I think it is better to test on local. So my question, is there a way I can test on my changes on the local? I couldn't find documentation on this. I would appreciate if anyone can share some info on this.

Thank you!

**Comment by tylerjw** [comment]
A big part of that is just getting in and reading the code and then finding some project to do with it. If you have a real use for the feature, it will be much easier for you to understand how it should work and how to test it.

**Comment by vdivakar** [comment]
Hi, I'm looking to start with a "good-first-issue". Could you share how can I verify or test the correctness of changes I made? I believe I would need to build moveit2 from source and check it's working after adding my changes. Is there a way to avoid this use GitHub checks instead?

@abhijelly I saw your pull request [2012](https://github.com/ros-planning/moveit2/pull/2012), could you share what steps did you take to make sure code runs fine after the change?

**Reply by AndyZe** [reply, marked as answer]
The options are:

- Install MoveIt2 from source. Directions here:  https://moveit.ros.org/install-moveit2/source/   That's what I do.
- Use the Rolling Docker image:  https://hub.docker.com/r/moveit/moveit2   I don't know too much about this.

Both ways  are fine. I'd say it really depends on:

- how familiar you are with Docker
- whether you have a nice, clean Ubuntu 22 installation. If you've modified your Ubuntu installation, then installing from source might be more difficult.

After I've built MoveIt, I usually run some of the tutorials to test things:  https://moveit.picknik.ai/main/doc/tutorials/tutorials.html
